mutt-alias.sh

```diff
@@ -114,34 +116,48 @@ for directory in "$@"; do
       out_age=$(( (now - out_date) / 86400 ))
 
       if [[ "$out_to" =~ ^${email_regexp}$ ]]; then
-        # Find previous entry's line number
-        prev_line_number="$(grep -F -i --max-count=1 "${out_to}" "${alias_file}")"
-        if { [ "0" = "$max_age" ] || [ "$out_age" -lt "$max_age" ]; } && [ "${prev_line_number}" = "" ]; then
+        # Find previous entry
+        grep -F -i -q "${out_to}" "${alias_file}" "${alias_file_new}"
+        grep_ret=$?
+        if { [ "0" = "$max_age" ] || [ "$out_age" -lt "$max_age" ]; } && [ "${grep_ret}" != "0" ]; then
           hr_out_date="$( date --date=@"$out_date" +%Y-%m-%d@%H:%M:%S )"
           new_entry="alias ${alias_to} $name_to <${out_to}> # mutt-alias: e-mail sent on ${hr_out_date}"
-          echo "${new_entry}" >> "${alias_file}"
+          echo "${new_entry}" >> "${alias_file_new}"
         fi
       fi
     done
     IFS=" "
   done
 done
 
+
 if perl -e 'use Encode::MIME::Header;' > /dev/null 2>&1; then
   perl -CS -MEncode -ne 'print decode("MIME-Header", $_)' \
-    "${alias_file}" > "${alias_file}.decoded"
-  mv "${alias_file}.decoded" "${alias_file}"
+    "${alias_file_new}" > "${alias_file_new}.decoded"
+  mv "${alias_file_new}.decoded" "${alias_file_new}"
 fi
 
 if [ "$filter" = "true" ]; then
   filter_regexp="<([[:alnum:]._%+-]*([0-9]{9,}|([0-9]+[a-z]+){3,}|\+|nicht-?antworten|ne-?pas-?repondre|not?([-_.])?reply|\b(un)?subscribe\b|\bMAILER\-DAEMON\b)[[:alnum:]._%+-]*\@([[:alnum:]-]+\.)+[[:alpha:]]{2,})> # mutt-alias: e-mail sent on"
   grep -Eiv \
     "$filter_regexp" \
-    "${alias_file}" > "${alias_file}.filtered"
+    "${alias_file_new}" > "${alias_file_new}.filtered"
+
+  mv "${alias_file_new}.filtered" "${alias_file_new}"
+
+  # Second round of filtering
+  filter_regexp="<([[:alnum:]._%+-]*\@noreply\..*)> # mutt-alias: e-mail sent on"
```

Comment:
It is not really clear what this second round achieves by this regexp. If you could help?

Reply:
Really sorry, it was a custom regex to filter "noreply" domains (in my case I was filtering the noreply message from Github).

I shouldn't committed that.